wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -553,18 +580,32 @@ private boolean checkPreconditionNotEmpty(Boolean isConditionSQL) {
    * @throws DirectiveLoadException
    * @throws DirectiveParseException
    */
-  private RecipeParser getRecipeParser(StageContext context)
-    throws DirectiveLoadException, DirectiveParseException {
+  private RecipeParser getRecipeParser(StageContext context) {
 
     registry = new CompositeDirectiveRegistry(SystemDirectiveRegistry.INSTANCE, new UserDirectiveRegistry(context));
-    registry.reload(context.getNamespace());
+    try {
+      registry.reload(context.getNamespace());
+    } catch (DirectiveLoadException e) {
+      LOG.error("Failed to reload the directive registry for namespace "
+              + "'{}' at stage '{}'. Please verify the namespace and ensure the directives are "
+              + "correctly configured. {}: {}", context.getNamespace(), context.getStageName(),
+          e.getClass().getName(), e.getMessage());
+      throw WranglerErrorUtil.getProgramFailureExceptionDetailsFromChain(e, null, ErrorType.USER);
```

Comment:
why are we logging if we are already throwing the exception?
This is an anti-pattern. Please do not log anywhere if we are already throwing the exception. Instead pass proper descriptive errorMessage in `ProgramFailureException`.

If you log the exception and then throw it, there's a chance that the exception will be logged again further up the call stack, resulting in two log events for the same error.

This comment applies to whole PR.